agent/stats/container.go

```diff
@@ -14,102 +14,68 @@
-
 	// SleepBetweenUsageDataCollection is the sleep duration between collecting usage data for a container.
-	SleepBetweenUsageDataCollection = 500 * time.Millisecond
+	SleepBetweenUsageDataCollection = 1 * time.Second
 
 	// ContainerStatsBufferLength is the number of usage metrics stored in memory for a container. It is calculated as
 	// Number of usage metrics gathered in a second (2) * 60 * Time duration in minutes to store the data for (2)
 	ContainerStatsBufferLength = 240
 )
 
-// ContainerStatsCollector defines methods to get container stats. This interface is defined to
-// make testing easier.
-type ContainerStatsCollector interface {
-	getContainerStats(container *CronContainer) (*ContainerStats, error)
+func newStatsContainer(dockerID string, client ecsengine.DockerClient) *StatsContainer {
+	ctx, cancel := context.WithCancel(context.Background())
+	return &StatsContainer{
+		containerMetadata: &ContainerMetadata{
+			DockerID: dockerID,
+		},
+		ctx:    ctx,
+		cancel: cancel,
+		client: client,
+	}
 }
 
-// LibcontainerStatsCollector implements ContainerStatsCollector.
-type LibcontainerStatsCollector struct{}
-
-// StartStatsCron starts a go routine to periodically pull usage data for the container.
-func (container *CronContainer) StartStatsCron() {
-	// Create the queue to store utilization data from cgroup fs.
+func (container *StatsContainer) StartStatsCollection() {
+	// Create the queue to store utilization data from docker stats
 	container.statsQueue = NewQueue(ContainerStatsBufferLength)
 
-	// Create the context to handle deletion of container from the manager.
-	// The manager can cancel the cronStats go routing by calling StopStatsCron method.
-	container.ctx, container.cancel = context.WithCancel(context.Background())
-	go container.cronStats()
+	go container.collect()
 }
 
-// StopStatsCron stops the periodic collection of usage data for the container..
-func (container *CronContainer) StopStatsCron() {
+func (container *StatsContainer) StopStatsCollection() {
 	container.cancel()
 }
 
-// newCronContainer creates a CronContainer object.
-func newCronContainer(dockerID string, dockerGraphPath string) *CronContainer {
-	statePath := filepath.Join(dockerGraphPath, DockerExecDriverPath, dockerID)
-
-	container := &CronContainer{
-		containerMetadata: &ContainerMetadata{
-			DockerID: dockerID,
-		},
-		statePath: statePath,
-	}
-
-	container.statsCollector = &LibcontainerStatsCollector{}
-	return container
-}
-
-// cronStats periodically pulls usage data for the container from cgroup fs.
-func (container *CronContainer) cronStats() {
+func (container *StatsContainer) collect() {
+	dockerID := container.containerMetadata.DockerID
 	for {
 		select {
 		case <-container.ctx.Done():
+			seelog.Debugf("Stopping stats collection for container %s", dockerID)
 			return
 		default:
-			stats, err := container.statsCollector.getContainerStats(container)
+			seelog.Debugf("Collecting stats for container %s", dockerID)
```

Comment:
Typically yes, though if there is a disconnect for any reason it'll get printed on reconnect as well.  I can move this to `TRACE` level if you think it's too noisy.